### PR TITLE
Adding service account feature

### PR DESCRIPTION
**Pull request by elhaus**
Implementation to get access tokens for service accounts.
#18

### CODE REVIEW

File: GMGoogleDrive/Public/Get-GDriveAccessToken.ps1 [hunk 1]
```diff
@@ -24,31 +36,111 @@
...
-        [string]$RefreshToken
+        [Parameter(ParameterSetName='ClientID', Mandatory, Position=2, ValueFromPipelineByPropertyName)]
+        [string]$RefreshToken,
+
+        [Parameter(ParameterSetName='P12ServiceAccount', Mandatory, ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName='JSONServiceAccount', Mandatory, ValueFromPipelineByPropertyName)]
+        [string]$PathServiceAccountFile,
+
+        [Parameter(ParameterSetName='P12ServiceAccount', ValueFromPipelineByPropertyName)]
+        #[SecureString]$CertificatePassword = (ConvertTo-SecureString "notasecret" -AsPlainText -Force),
```
Comment (elhaus) [inline GMGoogleDrive/Public/Get-GDriveAccessToken.ps1]: PSScriptAnalyzer sees this `(ConvertTo-SecureString "notasecret" -AsPlainText -Force)` as a security risk.
My implementation avoids the alert from PSScriptAnalyzer.
From my perspective it is not a security risk as everbody can find the password in the [Google documentation](https://support.google.com/cloud/answer/6158849?hl=en&sjid=10155945132668630004-EU#zippy=%2Cservice-accounts)

**Comment by MVKozlov** [comment]
@elhaus, I took a quick look at your implementation. Looks good.
a few comments about the code:
   - I prefer `$PSVersionTable.PSVersion.Major` style PS version checking rather than string interpretations
   - Have you checked that Get-JWTToken in this module will not conflict with Get-JWTToken from any other source? For this reason, I try to name internal functions so that they do not fully correspond to the ps style
   - I think there should be a method of loading private keys from `[byte[]]` or `[string]` and directly from `[System.Security.Cryptography.X509Certificates.X509Certificate2]$certificate` to allow it to be stored in an alternate secure storage

**Comment by elhaus** [comment]
I used `$PSVersionTable.PSVersion -lt "6.1"` as `Get-PfxCertificate -Password` was introduced in PowerShell 6.1. Otherwise I need to compare Major and Minor. I tried to use the same comparison for both parts.

`Get-JWTToken` is also used in [PowerApps.Administration.PowerShell](https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/get-jwttoken?view=pa-ps-latest). Should I change it to `Get-GDriveJWTToken`?

I think the best way to store certificates would be the windows certificate store. It might be complicated to implement this, especially for the json file, as we read additional information from this file. In any case, someone who has access to the runtime information might also be able to read the file from disk.

**Comment by MVKozlov** [comment]
if there will be parameter that accepts `[System.Security.Cryptography.X509Certificates.X509Certificate2]`, the `Get-PfxCertificate` will not be needed at all
```
NAME
    Get-PfxCertificate
OUTPUTS
    System.Security.Cryptography.X509Certificates.X509Certificate2
```
then we won't have to take on the conversion function.

I seriously think that maybe we shouldn’t take on the work with .p12 and .json files at all?
but just tell in the readme how to get a string or byte array with a private key from them.
and take upon yourself only the conversion of this string or bytes into a private key object

since `Get-GDriveJWTToken` is an internal function, it doesn’t really matter what it’s named, as long as an external one is not called instead, so let it be

**Comment by MVKozlov** [comment]
@elhaus, what you think about last commits ?

**Pull request merged**